### ISSUE TITLE
Homekit Secure Video

### ISSUE THREAD

**Issue by TechRemarker** [homebridge-plugins/homebridge-camera-ffmpeg#338]
In June, Apple announced the upcoming Homekit Secure Video feature where all video will also be backed up in iCloud. While I assume it will have to be finalized before we know for sure, but in theory will this plugin be able to support that, or do we think that feature will only work with manufacturer's working directly with Apple?

**Comment by IRONROCK20** [comment]
+1 It could be really cool!

**Comment by netRunner0** [comment]
News?
@KhaosT 😉

**Comment by IRONROCK20** [comment]
Yeah!!! That could be fantastic!

**Comment by Rocky1992** [comment]
HomeKit Secure Video requires a 200GB or 2TB iCloud storage plan and a home hub such as Apple TV, HomePod, or iPad. So I don't think it will need any further coding for this plugin.
Apple may have given the ability for HomeKit Secure Video just by installing iOS 13 with any camera viewed in home. We will see on 9.19.

**Comment by saschawerner** [comment]
From the screenshots so far, it requires a certified device.

**Comment by peros550** [comment]
Any news on this hot topic ?

**Comment by Velkas** [comment]
Count me in for interest on this.

**Comment by iRayanKhan** [comment]
Ok so here's my theory for why this isn't going toe implemented soon.
When you submit feedback for HomeKit in the feedback app (beta OS testers), it asks if you are in the HomeKit Beta Program for either TV's or one other thing. So far HomeKit Secure Video is not an option, but it has always been updated as soon as a new HomeKit feature has been announced/Ready to Test. Routers aren't on there either so I guess it may be a while for it to be open. Maybe it will be limited to first party HomeKit accessories due to the "HomeKit Chip" to make it secure. Just a theory, but corrected me if I'm wrong or have any other plausible theories.

**Comment by ageorgios** [comment]
iOS 13.2 is released with HomeKit Secure Video feature

**Comment by petergeneric** [comment]
The latest [HAP (Non-Commercial) documentation (R2)](https://developer.apple.com/homekit/specification/) still has no reference to HomeKit Secure Video

**Comment by netRunner0** [comment]
By the way, I found there the on / off characteristics of the night mode for the camera. Is it possible to add, in the form of transmission via mqtt?

**Comment by roysbike** [comment]
+1

**Comment by roysbike** [comment]
The latest HAP (Non-Commercial) documentation (R2) still has no reference to HomeKit Secure Video
[HAP-Specification-Non-Commercial-Version.pdf](https://github.com/KhaosT/homebridge-camera-ffmpeg/files/3804296/HAP-Specification-Non-Commercial-Version.pdf)

**Comment by mihailescu2m** [comment]
logitech circle 2 updated with secure video support

**Comment by iuliusmaximilian** [comment]
+1

**Comment by mikesalz** [comment]
Following...

**Comment by netRunner0** [comment]
Hap nodejs updated and now supports…
When to wait for support here?)

**Comment by jankais3r** [comment]
> Hap nodejs updated and now supports…
> When to wait for support here?)

HAP does not support Secure Video yet. See [here](https://github.com/KhaosT/HAP-NodeJS/issues/714).

**Comment by netRunner0** [comment]
@jankais3r !!!
https://github.com/KhaosT/HAP-NodeJS/pull/743

**Comment by Vannixxo** [comment]
Following this

**Comment by iRayanKhan** [comment]
I had a theory that only 1st party HomeKit accessories (verified and certified ones) would only be getting support for HomeKit secure video, due to the HomeKit chip, (I don't know if all the cameras have the chip). But, Hap does seem to have added HKSV characteristics, but not the full back end or logic of it. I suggested in ```homebridge-video-doorbell``` to use the HomeKit accessory Simulator in the latest version of Xcode to see if the simulator could provide any tips.

**Comment by mihailescu2m** [comment]
good news!
https://github.com/KhaosT/HAP-NodeJS/issues/741#issuecomment-573441082

**Comment by iRayanKhan** [comment]
Thank you for that link!

**Comment by mikeavdeev** [comment]
So what does it mean for this plugin and for us?

**Comment by iRayanKhan** [comment]
@mikeavdeev That HomeKit Secure Video may be coming to this plugin, and we can view the past and notifications and get motion detection. Search up HomeKit Secure Video demo on YouTube or something and that's what we'd get.

**Comment by Vannixxo** [comment]
This is going to be amazing... we could get full functionality without the use of other workarounds for notifications etc., can't wait!

**Comment by edwardbattistini** [comment]
This is really exciting.

**Comment by TheRealGreatOldOne** [comment]
Is this definitely an upcoming enhancement? If so, is there an eta as to when we’re likely to see it?

**Comment by netRunner0** [comment]
It's time ...

**Comment by Vannixxo** [comment]
> It's time ...

What?

**Comment by BBaoVanC** [comment]
> > It's time ...
> 
> What?

https://github.com/homebridge/homebridge/releases/tag/1.0.1

> Added new services and characteristics for:
>    * HomeKit Routers (WiFiRouter and WiFiSatellite services)
>    * HomeKit Secure Video (CameraOperatingMode and CameraEventRecordingManagement services)

**Comment by Vannixxo** [comment]
> > > It's time ...
> > 
> > 
> > What?
> 
> https://github.com/homebridge/homebridge/releases/tag/1.0.1
> 
> > Added new services and characteristics for:
> > 
> > * HomeKit Routers (WiFiRouter and WiFiSatellite services)
> > * HomeKit Secure Video (CameraOperatingMode and CameraEventRecordingManagement services)

Thanks dude didn’t get the notification! Wow

**Comment by BBaoVanC** [comment]
I'm excited for what this brings... hopefully we can have virtual HomeKit Secure Video cameras and/or the ability to port 3rd party cameras to support HomeKit Secure Video.

**Comment by KhaosT** [comment]
Sorry to be a downer but as of right now, there is no plan to support HomeKit Secure Video. Here are the reasons:
- HomeKit Secure Video requires the device to support motion detection (or doorbell button), which doesn't really fit a ffmpeg based plugin. Maybe something that's more specific to certain hardware will make more sense.
- The functionality provided by HomeKit Secure Video is really limited (just clips from detected motion). It's hard to justify all the complications need to go through to support that.

I'm happy to accept PR if someone is interested in bringing HomeKit Secure Video to this plugin, otherwise unless the functionality get enhanced significantly in the future iOS releases, it's unlikely for this plugin to support the feature.

**Comment by mneumark** [comment]
@KhaosT But you have support for using a motion sensor to trigger motion detection. Couldn't you trigger off that?

**Comment by netRunner0** [comment]
@KhaosT 
Many cameras have different capabilities for delivering motion detection information:
mqtt
http
Homekit motion sensor

I would implement (if I could) these features in the plugin, and users themselves will choose what they need

**Comment by toshibochan** [comment]
Look Samfox2 github have button and motion maybe can implement 
https://github.com/Samfox2/homebridge-videodoorbell

**Comment by codyc1515** [comment]
Just adding onto what @KhaosT said: It's also fairly expensive. In my country it's $16.99 a month to have more than one camera.

**Comment by mneumark** [comment]
I’m in the US. You only have to be paying $2.99/month for 200 GB of storage here. That’s not bad at all.

**Comment by codyc1515** [comment]
The 200GB plan only gives one camera. If you want more you need the most expensive plan. Even then I believe it’s called arbitrarily at five.

**Comment by snikch** [comment]
I don't think the expense is really the concern here. I want this feature *because I'm already paying for the storage so it's effectively free for me*.

**Comment by TheRealGreatOldOne** [comment]
> I don't think the expense is really the concern here. I want this feature _because I'm already paying for the storage so it's effectively free for me_.

Ditto.

**Comment by iRayanKhan** [comment]
I just want it because I want to be able to rewind recordings in the home app. Currently no other way to do this without HKSV right?

**Comment by stcbus** [comment]
Yeah I mostly want it for the UI and to easily get pet and person
notifications without relying on random services.

**Comment by alexmace** [comment]
We have a 2TB plan shared between the family, so we'd get 5 cameras for "free"

**Comment by martijndierckx** [comment]
I really want this as well.
The big added value here is that Apple will be able to turn random motion events (which might just be triggered by a moving tree branch) into real ones where notifications can for example only be shown when a person is detected.

**Comment by ChrisLinderwell** [comment]
+1 here

Any update on this one? Is it practical that this gets worked out - sounds like a lot of work

**Comment by hazcod** [comment]
Update: HSV is separate from the (free) HomeKit spec and only for commercial licencees..

**Comment by peros550** [comment]
Let’s hope they will announce something next week at WWDC

**Comment by ChrisLinderwell** [comment]
Did this get integrated or was there an announcement (maybe I missed) that enables this at WWDC?

**Comment by alexmace** [comment]
Why was this closed?

**Comment by mneumark** [comment]
Agreed. The keynote only gave more reason to want this feature.  Not less.

**Comment by donavanbecker** [comment]
sorry closed on accident.

**Comment by vkedwardli** [comment]
https://twitter.com/HerrSupereg/status/1214025078603755520
https://github.com/Supereg/secure-video-specification

There is an unofficial spec (in case you don't know)

**Comment by llemtt** [comment]
@vkedwardli those specifications confirm that it's actually beyond ffmpeg capabilities (HDS protocol)

another video processor must be used, maybe a custom version of ffmpeg or a pipe into another process implementing HDS (this could be done also inside the plugin eventually)

**Comment by hazcod** [comment]
I mentioned it in https://github.com/brutella/hc/issues/174
Imagine the possibilities :-)

**Comment by monza258** [comment]
Any news about secure video?

**Comment by jnanes** [comment]
Really interested in this. Is there somewhere I can donate?

**Comment by Supereg** [comment]
@california444 sorry for the delay. Seems interesting, I'm trying to test it this week.
If I'm finished with the hap-nodejs v0.8.0 beta update I hope I can get back to working on this. Maybe we can work on some sort of preliminary/beta API for secure video.

**Comment by california444** [comment]
HI,

in the meantime I (maybe) found an even better alternative, but hadn't time to test it, yet.
This Project [MP4Frag](https://github.com/yonigozman/mp4frag) pipes data and fragment mp4 into an initialization segment and media segments. That sounds promising.

If I find time to test I will share my result again...

**Comment by california444** [comment]
HI,

my test result with MP4Frag is uploaded [here](https://github.com/california444/homebridge-camera-ffmpeg/tree/master/homekitsecurevideo/mp4frag).

That looks even better and is a lot more convenient. MP4Frag buffers the lastest x segments as well as the initialisation data for you.
I configured the buffer size to two segments. After that I set up a callback for new segments as soon as they are completely available....

Generated files are:

- a full mp4 file (playable in Quicktime)
- a initialisation mp4 (contains ftyp and moof)
- an init + 2* 4s of buffer mp4
- 2 segments each appr. 4s long (moof and mdata, first frame is a keyframe)

Here is my ffmpeg command is used:
`const ffmpeg = spawn(
    'ffmpeg',
    ['-reorder_queue_size', '5', '-rtsp_transport', 'tcp', '-i', 'rtsp://url', '-c:v', 'copy', '-preset', 'veryfast', '-force_key_frames', '0:00:04', '-sc_threshold', '0', '-c:a', 'aac', '-b:a', '128k', '-ac', '2', '-f', 'mp4','-frag_duration', '4000000', '-fflags', '+genpts', '-movflags', 'empty_moov+default_base_moof', '-metadata', 'title="Homekit Stream"', '-reset_timestamps', '1', 'pipe:1'],
    {stdio: ['ignore', 'pipe', 'inherit']}
);`

**Comment by Wildcat1** [comment]
All, just catching up with this thread and with many cams to test against, getting HSV running with home bridge camera-ffmpeg would be FANTASIC ! @donavanbecker @KhaosT just wondering what the latest is from your input WRT being able to implement this ? Thanks all, very cool if this can work

**Comment by Sunoo** [comment]
@Wildcat1 To my knowledge, nobody has quite figured out HSV yet (though it seems we’re getting pretty close). It does seem like I should be able to get support added to this plugin once people get those details ironed out though.

**Comment by Supereg** [comment]
@Wildcat1 don't know if you discovered the issue already over at the hap-nodejs repo. If you didn't could also be interesting to read in regards to HomeKit Secure Video https://github.com/homebridge/HAP-NodeJS/issues/741

**Comment by roysbike** [comment]
I bought a xiaomi aqara  G2H camera with HSV . how can she help? Ready to sniff answers from her or connect to her via serial

**Comment by california444** [comment]
> @california444 sorry for the delay. Seems interesting, I'm trying to test it this week.
> If I'm finished with the hap-nodejs v0.8.0 beta update I hope I can get back to working on this. Maybe we can work on some sort of preliminary/beta API for secure video.

@Supereg are you generally working on HSV specific service implementations in HAP-NodeJS? I saw that some of the required services are not available, yet. Are you planning to integrate them in a future release?

**Comment by Supereg** [comment]
Once I'm through my exam phase and homebridge v1.2.0 (and hap-nodejs v0.8.0) is finished and released, my plan was to start working on a preliminary HSV implementation.

**Comment by bdruth** [comment]
@Supereg - do you have the code you hacked together to test out the HSV protocol pushed up anywhere? Might be helpful for some folks to get familiar with that so we can be in a better position to help?

**Comment by Finch106** [comment]
I purchased a few Eufy cameras this week that offer homkit/HSV so I can provide whatever you need as far as UI or whatever.

**Comment by hazcod** [comment]
@Finch106 do you mean srtsp? (Secure RTSP)
I suppose homekit *secure* video wouldn't stream plain RTSP over IP.

**Comment by llemtt** [comment]
homekit uses SRTP (Secure RTP) streaming, there's always a lot of confusion:

RTSP (and it's secure version SRTSP) it's an higher level streaming protocol that manages the "control plane" of the streaming (sessions setup, start/stop,...), it's the "streaming version" of HTTP. RTSP can support different "data plane" streaming protocols like RTP, TCP, HTTP, UDP,...

RTP/SRTP requires a separate "control plane" to work. Homekit "control plane" is embedded into HAP.

cheers

**Comment by rncry** [comment]
Following

**Comment by Wildcat1** [comment]
Question for all here. I'm running iOS 14 beta and tvOS beta. Even with the cams bridged through HomeBridge into HomeKit, I cannot display them on the latest tvOS beta. Anyone been able to test this yet ? Does it require HKSV to display on the APpleTV in tvOS beta. Trying to determine what the dependencies are in showing my cams that show fine in Home app (thanks to Homebridge + Camera-FFMPEG) but again NOT in the tvOS display camera area. Thanks

**Comment by Sunoo** [comment]
As far as I know, that shouldn’t require secure video. I honestly had completely forgotten that AppleTV could show cameras now, and never thought to test it. I do know the preview from the doorbell at least is working as expected.

**Comment by gondalez** [comment]
@Wildcat1 I have the latest beta and can confirm my cameras show in the home overlay (I'm using camera-ffmpeg also, not secure video).

Have you set your cameras as favourite in the home app on your phone? 
From memory only favourite cameras show on the ATV.

**Comment by iRayanKhan** [comment]
@Wildcat1 I have 4 Homebridge Cameras, and 1 HKSV Camera. The tvOS viewer seems to not work, and hasn't in any beta.

Look into the HomeCam app.

**Comment by BadgerBadgerAndFox** [comment]
Mixed bag here with ATV 4k and TVOS 14.
Ring running through homebridge does popup in ATV when pressed.
Home App in ATV displays all my cameras, some times they will work in liveview sometimes they wont, snapshot updates is also hit and miss, after they do start live feed returning to the tile view they do keep streaming, when they do start they also take an age to start. Full screen the resolution is not great. They are Unifi Protect cameras also integrated via Homebridge. I have a single old dlink cam configured directly in ffmpeg but its pretty old and the camera is failing, requiring frequent restarts, i'll see if i can get that up and report back.
For the protect cams i currently prefer the Unifi native ATV app.... would be good to see them working especially if they can be integrated as HKSV for enhanced alerting.

**Comment by hazcod** [comment]
Anyone that looked further into this now iOS 14 is out?

**Comment by lukasa1993** [comment]
is video playback only posible with secure video?

**Comment by Sunoo** [comment]
Yes.

**Comment by hackwell** [comment]
https://github.com/Supereg/secure-video-specification

**Comment by erokoder** [comment]
Any news about HSV with ffmpeg homebridge?

**Comment by Sunoo** [comment]
https://github.com/homebridge/HAP-NodeJS/issues/741#issuecomment-725154331

**Comment by Sunoo** [comment]
There is currently a PR to add Secure Video support to HAP-NodeJS. Once that gets accepted and support is added to Homebridge, I plan to add support to this plugin as well. Keeping this issue locked for now as I have no ETA on any of that.

https://github.com/homebridge/HAP-NodeJS/pull/904

**Comment by mrsbaaa** [comment]
Gents,

i am not sure if you figured out how to add secure videos support yet?

**Comment by hazcod** [comment]
@mrsbaaa It's merged https://github.com/homebridge/homebridge/pull/3056

**Comment by Sunoo** [comment]
Yes, currently my plan is to spend this weekend getting this working on this plugin, provided that life doesn’t get in the way.

**Comment by martijndierckx** [comment]
> @mrsbaaa It's merged in homebridge [homebridge/homebridge#3056](https://github.com/homebridge/homebridge/pull/3056)

Now that’s what I call good news!

**Comment by martijndierckx** [comment]
@Sunoo Since many camera's wont provide the needed i-frame every 4 seconds @ 1920x1080. Will we see a flag which will make ffmpeg force it, like it's being done in f.e. Scrypted?:
https://github.com/koush/scrypted/blob/d0988023950fc36532b7e8d8e1f885470f7526b8/plugins/homekit/src/types/camera.ts#L56

I know this will result in extra CPU overhead, but it's better than not having HSV :-)

f.e. My camera's output 1920 x 1088, with an i-frame every second in H264+AAC (via NX Witness)

**Comment by Sunoo** [comment]
@martijndierckx I assume I’ll have something like that if it’s actually required. I’ll need to do a lot of testing, and it’s likely the first release with HSV support will not catch 100% of the edge cases.

Also, I’m assuming that using the copy vcodec will be even more fragile than it is today. I may have to have a setting to ignore that for HSV video.

**Comment by martijndierckx** [comment]
Let me know if I can help you test. I'm very eager to get it working here :-)
I love NX Witness as an NVR, and frigate does a proper job of object-detection. But I miss the seamless integration in Homekit...

**Comment by Sunoo** [comment]
I will keep that offer in mind, thanks. I’ll have to see how things go, but I’m really hoping to push out at least a beta version this weekend.

**Comment by micheljansen** [comment]
Awesome. I'm also happy to help testing if that's useful.

**Comment by fate8383** [comment]
also ready to test to help any time.

**Comment by dswarne** [comment]
100% available for testing

**Comment by OfficialHoSay** [comment]
Will definitely help testing when a beta is available.

**Comment by martijndierckx** [comment]
@Sunoo any luck on getting a beta out on the hksv branch?

**Comment by adx74** [comment]
Yes i'm also ready to test the hksv branch, i already have homebridge v1.4.0-beta.7 installed.

**Comment by burnbrigther** [comment]
Ready!  Don't want to try camera.ui.  Are we close to beta release?

**Comment by nkpm** [comment]
Ready for testing as well. homebridge v1.4.0-beta.7 running well.

I tried installing with `npm install -g Sunoo/homebridge-camera-ffmpeg#hksv --save` but no look yet. 

Should it work already?

**Comment by Sunoo** [comment]
The HKSV branch is not in a working state quite yet. Sorry.

Also: For those who haven’t heard, I had a burst pipe in my house on Monday, which took out some of my cameras, the Raspberry Pis I use for development and testing, and my laptop power cord (among other things that don’t impact this). I’m hoping to get some sort of setup going soon so I can get back into it, hopefully this weekend, but things are a bit crazy over here right now.

**Comment by nkpm** [comment]
ok thanks for sharing so quickly.

I whish you all the best for recovery of your house!!

Beginning this year i switched to running all my homebridge stuff in docker containers for exactly this scary situation...

**Comment by Sunoo** [comment]
Thanks, fortunately I have insurance, so things should mostly work out in the end.

I had Docker around, to confirm my plugins worked under it, but I found no good way to rapidly test within it.

One I have a power cord again, I think my plan is to spin up a VM on my laptop and use that.

**Comment by mikesalz** [comment]
Maybe this is a dumb question - But once this awesomeness is complete, will plugins like Ring and Arlo be compatible with HomeKit Secure Video?

**Comment by nanosonde** [comment]
> Maybe this is a dumb question - But once this awesomeness is complete, will plugins like Ring and Arlo be compatible with HomeKit Secure Video?

From what I have seen so far it would require them to implement the additional delegates too.

Concerning the Ring Plugin: do not expect that it is going to be implemented by the original author: https://github.com/dgreif/ring/issues/177#issuecomment-916507211

**Comment by Sunoo** [comment]
Unfortunately it is something each plugin will need to implement on their own.

**Comment by nanosonde** [comment]
Concerning the ring plugin: https://github.com/dgreif/ring/issues/177#issuecomment-916507211

**Comment by nkpm** [comment]
@Sunoo i've used a docker test environment for react native development. How it worked there: code was "hot loaded" by putting it in a folder munted as volume. 

-> homebridge could be installed inside the container and your latest code could be in that folder. homebridge could be restarted everytime stuff changes in that folder...

i really enjoy being pretty independent of the host pi and run multiple versions of homebridge for different purposes (e.g. testing your ffmpeg beta) in parallel.

**Comment by burnbrigther** [comment]
Are we close? :)

**Comment by Sunoo** [comment]
It’s been close for a while now.

**Comment by nkpm** [comment]
quite an arc of tension 😳

**Comment by stcbus** [comment]
So does homebridge-camera-ui without installing a whole other automation system. 

–Steve

**Comment by steverae68** [comment]
Any news on  beta release for HKSV ?

**Comment by charlesmkelley** [comment]
Any news on a release that includes HKSV?

**Comment by sergoo-lbt** [comment]
Is the plugin dead?

**Comment by Sunoo** [comment]
No, my life has just been complicated recently.

**Comment by steverae68** [comment]
No problem - thanks for the update and hope everything works out in the end for you mate.

We’re here for ya if you need us

**Comment by nkpm** [comment]
+1

**Comment by sergoo-lbt** [comment]
> Нет, моя жизнь в последнее время была сложной.

Understood. Everything will get better! Take care of yourself.

**Comment by albinmedoc** [comment]
This feature is required for a good camera experience. How can we manage to implement this?

**Comment by albinmedoc** [comment]
We have this [example](https://github.com/homebridge/HAP-NodeJS/blob/b0d988b63c758d10157adfc74f865c8b807902d8/src/accessories/Camera_accessory.ts#L463) to get started with. 

I will give it a try.

**Comment by steverae68** [comment]
TBF - I’ve gave up on camera-mpeg and have switched to uCamera.Hope camera-mpeg gets updated soon as it’s way less resource hungry Sent from my iPhoneOn 24 Jul 2022, at 12:04, Albin Médoc ***@***.***> wrote:﻿
We have this example to get started with.
I will give it a try.

**Comment by scaryfast8750** [comment]
@Sunoo any updates on HKSV support?

**Comment by steverae68** [comment]
I think, by now we have to assume this plugin ain’t getting updated anymore.  Shame but life does have a habit of getting in the way with these things.I personally wish @sunoo every success and thank him for his years of support to this amazing plugin.As for HKSV, I have switched over to CameraUI - there are other solutions out there too such as Scrypted so all is not lost.  Sent from my iPhoneOn 12 Aug 2022, at 03:16, scaryfast8750 ***@***.***> wrote:﻿
@Sunoo any updates on HKSV support?

**Comment by Dids** [comment]
While I love CameraUI, it's also too resource and feature heavy for what I'd personally need it for, which is to simply expose camera streams to HomeKit via Homebridge, as well as have HKSV support.

Unless this gets updated, or a more viable lightweight alternative doesn't turn up, I might simply fork this project and add minimal HKSV support to it, as well as update and clean up the project a bit, then publish it as an experimental and minimal alternative.

(Note: the APIs required for HKSV aren't too complex, so it's fairly trivial to implement on top of this pligin, the hard part is that it requires enough time and testing to do right)

**Comment by nkpm** [comment]
@Dids I'm available for testing/qa.

**Comment by nanosonde** [comment]
> Unless this gets updated, or a more viable lightweight alternative doesn't turn up, I might simply fork this project and add minimal HKSV support to it, as well as update and clean up the project a bit, then publish it as an experimental and minimal alternative.

I am also still interested to enhance my PR https://github.com/Sunoo/homebridge-camera-ffmpeg/pull/1253 concerning two-way audio with support for HKSV in the future. So if there would be a new minimal cleaned-up stable working version with HKSV support, I would probably still be interested to add SIP two-way audio for it.
However, please note that personally, I will never create a pure SIP two-way audio video doorbell plugin that I have to maintain or support. The reason is that I do not have enough of the very valuable resource called "time".

**Comment by steverae68** [comment]
I have to agree with you that CameraUI is very resource intensive, however, it is the only plugin running on my particular pi so not to much of an issue…I’d be happy to test any fork of this if needed.Sent from my iPhoneOn 12 Aug 2022, at 08:04, Pauli Jokela ***@***.***> wrote:﻿
While I love CameraUI, it's also too resource and feature heavy for what I'd personally need it for, which is to simply expose camera streams to HomeKit via Homebridge, as well as have HKSV support.
Unless this gets updated, or a more viable lightweight alternative doesn't turn up, I might simply fork this project and add minimal HKSV support to it, as well as update and clean up the project a bit, then publish it as an experimental and minimal alternative.
(Note: the APIs required for HKSV aren't too complex, so it's fairly trivial to implement on top of this pligin, the hard part is that it requires enough time and testing to do right)

**Comment by scaryfast8750** [comment]
Agree with all...CameraUI is just too much stuff.  And I bet we arent alone, which is why this project(CameraFFMPEG) is so popular.

@Dids @nkpm @steverae68  I am also available for testing.

**Comment by scaryfast8750** [comment]
Anyone still willing to try to take this on and add to the code? 

Scrypted works fine, but I miss the simplicity of CameraFFMPEG.

**Comment by Sunoo** [comment]
I guess I sorta owe everyone a bit of an update. This has been a very difficult year for many reasons. I know this is somewhat going to sound like a list of excuses, but it's the situation:

- Near the start of the year, I had a pipe burst and do a lot of damage to my house (many of you will already be aware of this, but I'm including it anyways.) While reconstruction has finished, I'm still fighting with the insurance company over this claim.
- With the burst pipe, many of my IP cameras were destroyed. I do plan to replace them (and several other things), whenever the insurance is settled.
- Also near the start of the year, my small team at my day job had three people leave the company. This has drastically increased my workload there, and pushed me towards burning out a few times. Unfortunately, I need to prioritize my day job which pays my bills over personal work.
- About a month or so back my father was diagnosed with cancer.
- My cat has developed multiple health issues over the year, which has required a lot of care and semi-frequent vet visits.
- For reasons I will not go into publicly, I am currently starting to look for a new job.

All that said - I am still around here. I've been attempting to answer issue tickets here where I can and make sure at minimum any security issues are patched in my various projects as promptly as possible. It's just been a long time since I've had the energy to do much personal coding beyond that for a while now (I hope things are settling down soon though, knock on wood).

A few people mentioned they were possibly looking into adding support for HKSV themselves. If someone else gets HKSV working with this code base, I'd be happy to accept a pull request, or possibly even add a contributor onto this project.

I'm deeply sorry that this highly anticipated feature has been so delayed, and I probably should have posted this update a while ago, but I keep hoping things will settle down and let me focus on this stuff more. Unfortunately that just hasn't happened yet.

**Comment by evelant** [comment]
Please don't apologize, you've donated a ton of your time and talent to build and share this excellent project with the community for free! I for one am very grateful for that, thank you 🙏. I hope things start looking up for you. 

I don't have the time to hack on this right now, hopefully someone else can take up the torch until things get more stable for you.

**Comment by jankais3r** [comment]
Hi David,

Sorry to hear you've had a tough year. You don't owe us anything, and you certainly don't have to apologize when life gets in a way. Open source contribution in your free time should be about having fun, and that's not always possible due to external factors. Thank you for your work so far and good luck with your more pressing matters. Time to see if anyone else in the community can step up to the challenge :)

**Comment by scaryfast8750** [comment]
﻿
Dude. Takes courage to put it all out there like this so kudos to you and you should absolutely prioritize yourself and the things you care about in your life first. Not a bunch of us strangers using your free code!

You owe us nothing and we are all(at least me) for the time you do put in. Good luck with all the challenges and wish you the best.

**Comment by jlg89** [comment]
First priority is household and family, you're doing the right things. If you've felt any pressure from this community to put this project before more important things, then it's we who owe you an apology. Hang in there, be there for your dad. This code will be here whenever you get back around to it.

**Comment by steverae68** [comment]
Hey David, As I’ve said many times before, family and personal life always comes first.  It’s not like we don’t have alternatives to get HKSV and we will all happily use these until someone can pick up this project and push it forward.  I’m old and don’t have a clue where to even start with this as far a coding is concerned (I trained in COBOL and Basic in the 80’s - Showing my age here) - but I’m happy to test any beta builds with my setup if needed.Good luck David in your future and if you need to take time out and just chill on the toilet… che k out my iOS game - https://RaeEnterprises.UkThank you for all you time over these last few years and I echo everyone else’s comments in the community - take it easy Sent from my iPhoneOn 30 Aug 2022, at 15:03, jlg89 ***@***.***> wrote:﻿
First priority is household and family, you're doing the right things. If you've felt any pressure from this community to put this project before more important things, then it's we who owe you an apology. Hang in there, be there for your dad. This code will be here whenever you get back around to it.

**Comment by jlg89** [comment]
FWIW, I'm still using homebridge-camera-ffmpeg where HSV, motion detection, etc. aren't needed (KISS principle), but I just installed Scrypted using the docker method on rPi, moved my August Doorbell Cam Pro from homebridge-camera-ffmpeg (using motion and mqtt for motion detection) to Scrypted, with all its built-in stuff. It works beautifully, full HSV support, a very well-put-together system.

David, I don't see much reason for you to worry about HSV support unless it's just a burr under your saddle. Personally, if you have any dev time, I'd prefer you spend it on creating a Scrypted plugin version of homebridge-dafang-mqtt-republish.

UPDATE: With some help from the Scrypted dev, I was able to create an MQTT device script that provides pan/tilt switches for a Dafang Hacks camera to HomeKit via Scrypted. The script can easily be expanded to include whatever other controls are desired (LEDs, motion detection, etc.). I now have my old Dafang-hacked WyzeCams working with HSV, with pan/tilt control, in Scrypted.

The script is posted at https://github.com/koush/scrypted/issues/423#issue-1445939355

**Comment by derek-miller** [comment]
Any update on this issue? Would be great to have this functionality in this driver without the need for Scrypted.

**Comment by x23piracy** [comment]
Hi,

i HSV still missing for this project? I just came along from a link which mentions camera ffmpeg has HSV support.
Can someone clearify please?


Best Regards X23

**Comment by 350d** [comment]
Just a yearly reminder about HKSV status here. The only missed feature here on this project so far.  Hope you doing well these days @Sunoo

**Comment by slyoldfox** [comment]
> Just a yearly reminder about HKSV status here. The only missed feature here on this project so far. Hope you doing well these days @Sunoo

I subscribed here because I wanted to implement HKSV on my (rooted) Bticino doorbell, so I was looking for a HSKV implementation at that time.

I am unsure what is lacking from the Homebridge side (I use HA mainly with my now Bticino HKSV compliant doorbell).
It wasn't that hard to implement and I think most of the code I copy/pasted from this project. 

If anyone can point out what is lacking, I don't mind having a look (but if it's UI stuff .. that's not my thing).

**Comment by Sunoo** [comment]
> Just a yearly reminder about HKSV status here. The only missed feature here on this project so far. Hope you doing well these days @Sunoo

I’ve continued being very busy, unfortunately. I have, however, transferred ownership of this repository over to the Homebridge organization. I’m sticking around, but I’m hoping some of them can provide some of the time I haven’t been able to lately.

**Comment by donavanbecker** [comment]
If anyone is interested. We have HKSV published to the `alpha` tag.

if anyone would like to help implement this support, and improve what `sunoo` has already implemented. Please open a PR.

**Comment by 350d** [comment]
Just in case - old HKSV PR [available here ](https://github.com/homebridge-plugins/homebridge-camera-ffmpeg/pull/1200)

**Comment by donavanbecker** [comment]
@350d that is what is merged into alpha

**Comment by steverae68** [comment]
I’ll fire up Homebridge again, haven’t used it for years….. and test this outI’ll let you know Sent from my iPhoneOn 28 Oct 2024, at 16:10, Donavan Becker ***@***.***> wrote:﻿
@350d that is what is merged into alpha

**Comment by donavanbecker** [comment]
v4.0.0 has been released.
